projects/RabbitMQ.Client/client/impl/Connection.Receive.cs

```diff
@@ -66,57 +66,68 @@ private void MainLoop()
             FinishClose();
         }
 
-        private void ReceiveLoop()
+        private async Task ReceiveLoop()
         {
             while (!_closed)
             {
-                InboundFrame frame = _frameHandler.ReadFrame();
+                while (_frameHandler.TryReadFrame(out InboundFrame frame))
+                {
+                    NotifyHeartbeatListener();
+                    ProcessFrame(frame);
+                }
+
+                // Done reading frames synchronously, go async
+                InboundFrame asyncFrame = await _frameHandler.ReadFrameAsync().ConfigureAwait(false);
```

Comment:
Nit suggestion: 

```suggestion
                InboundFrame asyncFrame = await _frameHandler.ReadFrameAsync()
                   .ConfigureAwait(false);
```

Reply:
You are welcome to ignore. This is just my taste. I think it separates clearly the actual async operation from the `ConfigureAwait` noise.

Reply:
I like the suggestion. I think we'll address this in bulk before 7.0 ships.